pkg/util/helper/luakube.go

```diff
@@ -0,0 +1,109 @@
+package helper
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util/lifted"
+)
+
+// GeneratePodFromTemplate generate a simple pod object from a template.
+func GeneratePodFromTemplateAndNamespace(template *corev1.PodTemplateSpec, namespace string) *corev1.Pod {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+		},
+	}
+
+	pod.Spec = *template.Spec.DeepCopy()
+	return pod
+}
+
+// GetDependenciesFromPodTemplate returns DependentObjectReferences according to podObj.
+// It is from `pkg\resourceinterpreter\defaultinterpreter\dependencies.go`.
+func GetDependenciesFromPodTemplate(podObj *corev1.Pod) ([]configv1alpha1.DependentObjectReference, error) {
```

Comment:
You can directly use the `GetDependenciesFromPodTemplate` method in the `pkg\resourceinterpreter\defaultinterpreter\dependencies.go` file and set it as the export method.

Reply:
Perhaps we can move this function to the `pkg/util/helper/pod.go` file.

Reply:
+1